File: Felino.py
class Felino:
    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad
    
    def comer(self):
        print(f"{self.nombre} está comiendo")
    
    def dormir(self):
        print(f"{self.nombre} está durmiendo")

class Gato(Felino):
    def __init__(self, nombre, edad, raza):
        super().__init__(nombre, edad)
        self.raza = raza
    
    def maullar(self):
        print(f"{self.nombre} dice: Miau!")
    
    def jugar(self):
        print(f"{self.nombre} está jugando")

if __name__ == "__main__":
    mi_gato = Gato("Luna", 3, "Persa")
    
    mi_gato.maullar()
    mi_gato.comer()
    mi_gato.jugar()
    mi_gato.dormir()
    
    print(f"Nombre: {mi_gato.nombre}")
    print(f"Edad: {mi_gato.edad} años")
    print(f"Raza: {mi_gato.raza}")

File: Felino.java
class Felino {
    String nombre;
    int edad;
    
    public Felino(String nombre, int edad) {
        this.nombre = nombre;
        this.edad = edad;
    }
    
    public void comer() {
        System.out.println(nombre + " está comiendo");
    }
    
    public void dormir() {
        System.out.println(nombre + " está durmiendo");
    }
}

class Gato extends Felino {
    String raza;
    
    public Gato(String nombre, int edad, String raza) {
        super(nombre, edad);
        this.raza = raza;
    }
    
    public void maullar() {
        System.out.println(nombre + " dice: Miau!");
    }
    
    public void jugar() {
        System.out.println(nombre + " está jugando");
    }
}

class Main {
    public static void main(String[] args) {
        Gato miGato = new Gato("Mimi", 2, "Electrico");
        
        miGato.maullar();
        miGato.comer();
        miGato.jugar();
        miGato.dormir();
        
        System.out.println("Nombre: " + miGato.nombre);
        System.out.println("Edad: " + miGato.edad + " años");
        System.out.println("Raza: " + miGato.raza);
    }
}